src/data_request.rs

```diff
@@ -25,54 +59,67 @@ impl DataRequest {
         DataRequest { layers: b, lib: b, groups: b, kerning: b, features: b, data: b, images: b }
     }
 
-    /// Returns a `DataRequest` requesting all UFO data.
+    /// Returns a [DataRequest] requesting all UFO data.
     pub fn all() -> Self {
         DataRequest::from_bool(true)
     }
 
-    /// Returns a `DataRequest` requesting no UFO data.
+    /// Returns a [DataRequest] requesting no UFO data.
     pub fn none() -> Self {
         DataRequest::from_bool(false)
     }
 
-    /// Request that returned UFO data include the glyph layers and points.
+    /// Returns a mutable reference to a [DataRequest] that includes
+    /// glyph layers and points if `b` is `true` or excludes these
+    /// data if `b` is `false`.
```

Comment:
Rust doc comments work a lot like git commits: the first line is used as a short summary (and is shown beside an item in the module level docs), and then anything after the first newline is "additional detail" and is only shown on the docs page for that specific item.

So basically: one short line that is a summary, and then anything else below the fold.

In this case though, I'm not sure that the new docs are an improvement on the previous ones? In particular I don't think it is necessary to mention that this returns `&mut Self`. This pattern should be familiar to the caller from `std` (see [`std::process::Command`](http://doc.rust-lang.org/1.56.0/std/process/struct.Command.html#method.stdin)). If the previous docstring was too vague, I might instead say,

```
/// Specify whether layers (and glyphs) should be loaded.
```

Reply:
Noted. I will have a look at this one tonight and respond.

Reply:
Thanks for this information Colin.  I agree with your comments here and revised the initial draft in https://github.com/linebender/norad/pull/197/commits/fb78ef7c09a22e7fb941899726499a7b4f3cbe08

Please let me know if you feel that this improves the documentation in the section.

Reply:
Related:

I've tended towards adding a `# Notes` header in documentation sections that have longer and/or multi-issue text content below the summary line.  IMO this sets it off nicely in the HTML site, but this is very subjective.  If you feel that this leads to unnecessary header clutter in the docs please let me know.  I should have a full first draft available by the end of the day so that we can have a look.

Reply:
I think that `std` is a great guideline, and we should try to match the conventions used there. They don't seem to use `# Note` as a header very often, but they do often have lines that begin with 'Note:', for providing information that is not important in the general case, but is important to a deeper understanding of the API: see for instance the docs for [`String::drain`](http://doc.rust-lang.org/1.56.0/std/string/struct.String.html#method.drain).

I do *think* I've seen `# Note:` before, but generally for a longform exposition of some idea or concept, but I can't find an example. The best I can come up with is the [A ghastly note 👻👻👻](http://doc.rust-lang.org/1.56.0/std/marker/struct.PhantomData.html#a-ghastly-note-) section of `PhantomData`.

Reply:
Tyvm!  I'll do another pass through all of the new docs and remove unnecessary Note headers that do not follow this guideline.

Reply:
This is how it appears in the HTML

<img width="1510" alt="2021-11-10_13-23-59" src="https://user-images.githubusercontent.com/4249591/141171418-4586c74c-41f7-4bea-a5f3-053e6cf558fa.png">

Reply:
vs. this after https://github.com/linebender/norad/pull/197/commits/314e2168eb3b9ab6eb846248cc990ef70038821a

<img width="1510" alt="2021-11-10_13-28-56" src="https://user-images.githubusercontent.com/4249591/141172158-17e3b0c2-01ba-406d-ae1c-34ed3fae90e7.png">